File: M1/C/fact.c
#include <stdio.h>


int fact(int n){
    int res = 1;
    for (int i = 1; i < n; i++)
    {
        res *= i;
    }
    return res;  
}

int main(){
    int x = fact(5);
    printf("%d", x);
    return 0;
}

File: M1/Rust/src/bin/fact.rs
/* use std::io; */


fn fact(n:i32) -> i32{
    let mut res:i32 = 1;
    for _i in 1..n {
        res *= _i;
    }
    return res;
}

fn main(){
    let x:i32  = fact(13);
    println!("{}", x);
}